### PR TITLE
Try to send only one request to VMSS Azure API from nodepool handler

### PR DESCRIPTION
**Pull request by fiunchinho**
I realized that we were sending several different requests from the `nodepool` handler to fetch the same VMSS. I tried to fetch it once and pass it around so we can save some requests on every reconciliation loop.

I haven't tested it yet. What do you think?

### CODE REVIEW

File: service/controller/resource/nodepool/create_deployment_uninitialized.go
Comment (whites11) [inline service/controller/resource/nodepool/create_deployment_uninitialized.go:197]: vmss can be nil, does this even compile?

Comment (whites11) [reply service/controller/resource/nodepool/create_deployment_uninitialized.go:197]: I mean, shouldn't it be a pointer?

Comment (whites11) [reply service/controller/resource/nodepool/create_deployment_uninitialized.go:197]: Oh, now I got how it works.
Maybe I would use a pointer and set it to nil in case of 404, then check for pointer != nil where you want to use the `vmss`.
But I like the idea in general

Comment (tuommaki) [reply service/controller/resource/nodepool/create_deployment_uninitialized.go:197]: If we can use non-pointer value (especially given that e.g. autorest `Response` provides functionality), I feel it's better. Less unnecessary nil checks and less potential for null pointer dereferencing, but your call. I'm not blocking on this.

File: service/controller/resource/nodepool/create_deployment_uninitialized.go
Comment (tuommaki) [inline service/controller/resource/nodepool/create_deployment_uninitialized.go:163]: Do we need the vmss clients when we pass `vmss` instead?

Comment (fiunchinho) [reply service/controller/resource/nodepool/create_deployment_uninitialized.go:163]: Good point, I'll remove it.

File: service/controller/resource/nodepool/create_deployment_uninitialized.go
Comment (fiunchinho) [reply service/controller/resource/nodepool/create_deployment_uninitialized.go:197]: Yeah, I'd rather not to deal with pointers.

**Review by tuommaki — APPROVED** [review]

**Pull request merged**
